File: Front_Lisam_React/src/models/project.ts
import Client from "./client";
import {statut} from '../models/statut';


export default class Project{
    
    ProjectId: number ;
    SignatureDate : Date;
    ProjectNumber: string;
    ProjectLeader: string;
    Statut: statut;
    IsActive: boolean;
    ClientId: number |null;
    Client: Client;

    constructor(
        projectId: number = 0,
        signatureDate: Date = new Date(),
        projectNumber: string = "no value",
        projectLeader: string = "no value",
        statut: statut = 0,
        isActive: boolean = true,
        clientId: number = 0,
        client: Client = new Client()
    ) {
        this.ProjectId = projectId;
        this.ProjectLeader = projectLeader; 
        this.ProjectNumber = projectNumber;
        this.SignatureDate = signatureDate;
        this.Statut = statut;
        this.IsActive = isActive;
        this.ClientId = clientId; 
        this.Client = client;
    }
}

File: API_Lisam/API_Lisam/Models/Project.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace API_Lisam.Models
{
    public enum Statuts
    {
        Open = 1,
        Signed = 2,
        Factured = 3,
        Work_In_Progress = 4,
        Closed = 5
    }
    public class Project
    {
        public int ProjectId { get; set; }
        public DateTime? SignatureDate { get; set; }
        public string ProjectNumber { get; set; }
        public string ProjectLeader { get; set; }
        public Statuts? Statut { get; set; }
        public bool IsActive { get; set; }

        public int? ClientId { get; set; }
        public Client Client { get; set; }
    }
}